spec/services/participant_event_aggregator_service_spec.rb

```diff
@@ -17,25 +17,32 @@ def count_something_else(*)
 RSpec.describe ParticipantEventAggregator do
   let(:lead_provider) { create(:lead_provider) }
 
-  before do
-    10.times do
-      pr = create(:participation_record, lead_provider: lead_provider)
-      pr.join!
-      create(:participation_record, lead_provider: lead_provider)
+  context "event declarations" do
+    before do
+      10.times do
+        participant_declaration = create(:participant_declaration, lead_provider: lead_provider)
+        create(:participant_declaration, early_career_teacher_profile: participant_declaration.early_career_teacher_profile, lead_provider: lead_provider)
+        participant_record = create(:participation_record, lead_provider: lead_provider)
+        participant_record.join!
+      end
     end
-  end
 
-  context ".call" do
-    it "returns a count of the active participants" do
-      expect(described_class.call({ lead_provider: lead_provider })).to eq(10)
-    end
+    context ".call" do
+      it "returns a count of the active participants" do
+        expect(described_class.call({ recorder: ParticipationRecord, lead_provider: lead_provider })).to eq(10)
+      end
 
-    it "can be injected with a different recorder" do
-      expect(described_class.call({ recorder: DummyRecorder, lead_provider: lead_provider })).to eq(100)
-    end
+      it "returns a count of the unique start events" do
+        expect(described_class.call({ lead_provider: lead_provider })).to eq(10)
+      end
+
+      it "can be injected with a different recorder" do
+        expect(described_class.call({ recorder: DummyRecorder, lead_provider: lead_provider })).to eq(100)
+      end
 
-    it "can be injected with a different recorder and different scope" do
-      expect(described_class.call({ recorder: DummyRecorder, start: :count_something_else, lead_provider: lead_provider })).to eq(40)
+      it "can be injected with a different recorder and different scope" do
+        expect(described_class.call({ recorder: DummyRecorder, start: :count_something_else, lead_provider: lead_provider })).to eq(40)
+      end
```

Comment:
I am missing some context here.
I think `.call` should be in a `describe` block, and it should have one `context` block for each type of recorder that we can give it. I struggle to understand the difference between the first two tests from test descriptions.

Reply:
OK. Done as per offline conversation.